_deploy/r/gnoswap/gnft/gnft.gno

```diff
@@ -148,75 +197,121 @@ func Render(path string) string {
 	}
 }
 
-// Util
-func assertIsAdmin(address std.Address) {
-	if address != admin {
-		panic(addDetailToError(
-			errNoPermission,
-			ufmt.Sprintf("gnft.gno__assertIsAdmin() || only admin(%s) can call this function, called from %s", admin.String(), address.String()),
-		))
-	}
-}
+// SetTokenURI sets the metadata URI using a randomly generated SVG image
+// Parameters:
+//   - tid: The token ID to set the URI for
+//
+// Returns:
+//   - string: The token URI
+//   - error: Returns an error if the URI setting fails, nil otherwise
+func SetTokenURI(tid grc721.TokenID) (string, error) {
+	common.IsHalted()
 
```

Comment:
Please review if the address that made the request for tid (prev.Realm.Addr()) is the owner of tid and return early

Reply:
ffd80ea1e3fce168279ecb1797009b909da622fd